src/transformers/modeling_utils.py

```diff
@@ -1347,13 +1350,17 @@ def load(module: nn.Module, prefix=""):
     def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=False):
         module_keys = set([".".join(key.split(".")[:-1]) for key in names])
 
+        # torch.nn.ParameterList is a special case where two parameter keywords
+        # are appended to the module name, *e.g.* bert.special_embeddings.0
+        module_keys = module_keys.union(set([".".join(key.split(".")[:-2]) for key in names if key[-1].isdigit()]))
+
         retrieved_modules = []
         # retrieve all modules that has at least one missing weight name
         for name, module in self.named_modules():
             if remove_prefix:
                 name = ".".join(name.split(".")[1:]) if name.startswith(self.base_model_prefix) else name
             elif add_prefix:
-                name = ".".join([self.base_model_prefix, name])
+                name = ".".join([self.base_model_prefix, name]) if len(name) > 0 else self.base_model_prefix
```

Comment:
weird edge case in which the module is just an empty string `''` which will then lead to `bert.` instead of the correct `bert` key